src/backend/mailbox_manager/utils/dimail.py

```diff
@@ -163,3 +166,50 @@ def send_new_mailbox_notification(self, recipient, mailbox_data):
                 recipient,
                 exception,
             )
+
+    def synchronize_mailboxes_from_dimail(self, domain):
+        """Synchronize mailboxes from dimail - open xchange to our database.
+        This is useful in case of acquisition of a pre-existing mail domain.
+        Mailboxes created here are not new mailboxes and will not trigger mail notification."""
+
+        try:
+            response = session.get(
+                f"{self.API_URL}/domains/{domain.name}/mailboxes/",
+                headers=self.get_headers(),
+                verify=True,
+                timeout=10,
+            )
+        except requests.exceptions.ConnectionError as error:
+            logger.error(
+                "Connection error while trying to reach %s.",
+                self.API_URL,
+                exc_info=error,
+            )
+            raise error
+
+        if response.status_code != status.HTTP_200_OK:
+            return self.pass_dimail_unexpected_response(response)
+
+        dimail_mailboxes = ast.literal_eval(
+            response.content.decode("utf-8")
+        )  # format output str to proper list
+
+        people_mailboxes = models.Mailbox.objects.filter(domain=domain)
+        imported_mailboxes = []
+        for dimail_mailbox in dimail_mailboxes:
+            if not dimail_mailbox["email"] in [
+                str(people_mailbox) for people_mailbox in people_mailboxes
+            ]:
+                # creates a mailbox on our end
+                mailbox = models.Mailbox.objects.create(
+                    first_name=dimail_mailbox["givenName"],
+                    last_name=dimail_mailbox["surName"],
+                    local_part=dimail_mailbox["email"].split("@")[0],
```

Comment:
Est-ce qu'on ne devrait pas vérifier que la deuxième partie correspond bien au domain name ?

Reply:
Can you consider using `from email.headerregistry import Address` here to prevent any CVE (I did the same mistake elsewhere)? I know we should trust dimail but anyway I think it's safer to use battle-tested Python helpers here, just in case.

Reply:
Bonne question. Je préfèrerais qu'on laisse dimail tester de son côté que son API fonctionne bien comme il faut.

Si tu veux vérifier si le fix a été poussé, tu peux vérifier sur la prod de dimail que betagouv ne renvoie bien que des adresses betagouv (ou dans leur codebase que le test existe bien de leur côté)

Mais si tu veux faire ceinture+bretelle, on peut ajouter un filtre ligne 196 pour ne garder que les dimail_mailboxes qui correspondent au domaine.